data-pipeline/src/pipeline/mappings.py

```diff
@@ -14,11 +14,14 @@ def map_ofsted_rating(rating: str):
             return rating
 
 
-def map_phase_type(establishment_code: int, phase_code: int, provision: str):
-
-    if (pd.isna(establishment_code) or pd.isna(phase_code) or pd.isna(provision)):
+def map_phase_type(
+    establishment_code: int,
+    phase_code: int,
+    provision: str | None = None,
+):
+    if pd.isna(establishment_code) or pd.isna(phase_code) or pd.isna(provision):
```

Comment:
Left in place as it's out-of-scope but is this correct?

If _any_ of them are null, return null? If _all_ are null, I could see that but wouldn't the preference be to rely on at least one of them?

Reply:
Agree - I'm not sure why we need to rely on all 3.

Reply:
I think this method needs rewriting. I've done a bit of analysis on the GIAS data, and the method can be simplified to the below.

We'll require the `PhaseOfEducation` for all cases, but only `TypeOfEstablishment` in two cases. (Apologies for the C#)

````
string overallPhase = phaseOfEducation switch
{
    0 => GetNotApplicablePhases(typeOfEstablishment),
    1 => "Nursery",
    2 or 3 => "Primary",
    4 or 5 => GetSecondaryPhases(typeOfEstablishment),
    6 => "Post-16",
    7 => "All-through",
    _ => "Unkown"
};

string GetSecondaryPhases(int type)
{
    string result = type switch
    {
        40 => "University Technical College",
        _ => "Secondary"
    };

    return result;
}

string GetNotApplicablePhases(int type)
{
    string result = type switch
    {
        14 => "Pupil Referral Unit",
        39 or 45 or 46 => "Special",
        38 or 42 or 43 => "Alternative Provision",
        _ => "Unknown"
    };

    return result;
}
```

Reply:
@J05h-L is there any reason why the above wouldn't work for the Overall Phases?

Reply:
I don't think this would account for PhaseOfEducation (code) == 0 and TypeOfEstablishment (code) not in our checks. Then we should use Type of Provision - Phase from the academy_master_list or OverallPhase from the maintained_schools_master_list which we won't here

Reply:
@jrabbott, updated to reflect the above.

Reply:
LGTM